sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -204,32 +220,111 @@ public static Write write() {
+    }
+
+    /** Specifies the Cloud Spanner configuration. */
+    public ReadAll withSpannerConfig(SpannerConfig spannerConfig) {
+      return toBuilder().setSpannerConfig(spannerConfig).build();
+    }
+
+    /** Specifies the Cloud Spanner project. */
+    public ReadAll withProjectId(String projectId) {
+      return withProjectId(ValueProvider.StaticValueProvider.of(projectId));
+    }
+
+    /** Specifies the Cloud Spanner project. */
+    public ReadAll withProjectId(ValueProvider<String> projectId) {
+      SpannerConfig config = getSpannerConfig();
+      return withSpannerConfig(config.withProjectId(projectId));
+    }
+
+    /** Specifies the Cloud Spanner instance. */
+    public ReadAll withInstanceId(String instanceId) {
+      return withInstanceId(ValueProvider.StaticValueProvider.of(instanceId));
+    }
+
+    /** Specifies the Cloud Spanner instance. */
+    public ReadAll withInstanceId(ValueProvider<String> instanceId) {
+      SpannerConfig config = getSpannerConfig();
+      return withSpannerConfig(config.withInstanceId(instanceId));
+    }
+
+    /** Specifies the Cloud Spanner database. */
+    public ReadAll withDatabaseId(String databaseId) {
+      return withDatabaseId(ValueProvider.StaticValueProvider.of(databaseId));
+    }
+
+    /** Specifies the Cloud Spanner database. */
+    public ReadAll withDatabaseId(ValueProvider<String> databaseId) {
+      SpannerConfig config = getSpannerConfig();
+      return withSpannerConfig(config.withDatabaseId(databaseId));
+    }
+
+    @VisibleForTesting
+    ReadAll withServiceFactory(ServiceFactory<Spanner, SpannerOptions> serviceFactory) {
+      SpannerConfig config = getSpannerConfig();
+      return withSpannerConfig(config.withServiceFactory(serviceFactory));
+    }
+
+    public ReadAll withTransaction(PCollectionView<Transaction> transaction) {
+      return toBuilder().setTransaction(transaction).build();
+    }
+
+    @Override
+    public PCollection<Struct> expand(PCollection<ReadOperation> input) {
+      PCollection<ReadOperation> reshuffled = input
+          .apply("Pair wth random key",
+              WithKeys.of(new SerializableFunction<ReadOperation, String>() {
+
+                @Override public String apply(ReadOperation input1) {
+                  return UUID.randomUUID().toString();
+                }
+              }))
+          .apply("Reshuffle", Reshuffle.<String, ReadOperation>of())
+          .apply("Strip keys", Values.<ReadOperation>create());
+      List<PCollectionView<Transaction>> sideInputs = Collections.emptyList();
+      if (getTransaction() != null) {
+        sideInputs = Collections.singletonList(getTransaction());
+      }
+      NaiveSpannerReadFn fn = new NaiveSpannerReadFn(getSpannerConfig(), getTransaction());
+      return reshuffled.apply("Execute all query", ParDo.of(fn).withSideInputs(sideInputs));
+    }
+  }
+
+    /**
+   * A {@link PTransform} that reads data from Google Cloud Spanner.
```

Comment:
Usually we document this as "Implementation of {@link #read}" or something like that, to avoid redundancy.